fix(service-worker): allow creating post api requests after cache failure

apocalyp0sys opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Before creating a mutating http request, service-worker invalidates lru cache entry and writes to cache storage. Therefore, cache storage failure can prevent making post requests.


Issue Number: #33793


## What is the new behavior?

Cache storage error is catched and logged when trying to update lru cache table.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

apocalyp0sys committed:
refactor(service-worker): set http method in MockRequest constructor

Enable to create mocks for mutating http requests in tests
to check more scenarios.

gkalpak commented on packages/service-worker/worker/src/data.ts:
```diff
@@ -275,7 +275,15 @@ export class DataGroup {
+    } catch (err) {
+      // Writing lru cache table failed. This could be a result of a full storage.
+      // Continue serving clients as usual.
+      this.debugHandler.log(err, `DataGroup(${this.config.name}@${this.config.version}).syncLru`);
```
Nit: Consider changing `.syncLru` to `.syncLru()` to keep it similar to other similar debug logs.

gkalpak reviewed (approved):
Thx, @apocalyp0sys ❤️ 
(Nice commit message, btw 👍)

apocalyp0sys committed:
fix(service-worker): allow creating post api requests after cache failure

Before creating a mutating http request, service-worker
invalidates lru cache entry and writes to cache storage.
Therefore, cache storage failure can prevent making post requests.
Fix this by catching and logging cache error, add a test case.

Fixes #33793

gkalpak commented:
**merge-assistance:** The `integration_test` failure is unrelated.

matsko commented:
@apocalyp0sys please update the payload size amount so we can merge this.

apocalyp0sys commented:
@matsko rebased onto master
